### PR TITLE
Fix acl sync to properly filter default deny acls.

### PR DESCRIPTION
**Pull request by npinaeva**
`ExternalIDs[defaultDenyPolicyTypeACLExtIdKey]` is also used by multicast acls (despite the name), but multicast acl have different priorities.
It was not a bug, because multicast acls are updated before default deny, therefore only default deny acls will match the criteria. This commit just makes the filtering more specific.

### CODE REVIEW

**Comment by npinaeva** [comment]
`OVN Egress Gateway Operations [0m[0mhybrid route policy operations in lgw mode [0m[91m[1m[It] should create a single policy for concurrent addHybridRoutePolicy for the same node` failed

File: go-controller/pkg/ovn/external_ids_syncer/acl/acl_sync.go [hunk 1]
```diff
@@ -363,7 +363,8 @@ func (syncer *aclSyncer) updateStaleDefaultDenyNetpolACLs(legacyACLs []*nbdb.ACL
 	deleteOps []libovsdb.Operation, err error) {
 	for _, acl := range legacyACLs {
 		// sync default Deny policies
-		if acl.ExternalIDs[defaultDenyPolicyTypeACLExtIdKey] == "" {
+		if acl.ExternalIDs[defaultDenyPolicyTypeACLExtIdKey] == "" || acl.Priority == types.DefaultMcastDenyPriority ||
+			acl.Priority == types.DefaultMcastAllowPriority {
```
Comment (jcaamano) [inline go-controller/pkg/ovn/external_ids_syncer/acl/acl_sync.go:367]: I am confused here. Isn't the intention not to match multicast policies and make the filter more specific as stated in the PR message? Isn't OR'ing making the filter less specific? 

Did you mean to say multicast ACLs also use `defaultDenyPolicyTypeACLExtIdKey` external Id but with a different value?

Maybe add a comment explaining what we intend to match?

Comment (npinaeva) [reply go-controller/pkg/ovn/external_ids_syncer/acl/acl_sync.go:369]: Added a comment, hopefully it helps. 
To clarify, this `if` filters out not default deny acls, it continues with `// not default deny policy` comment. So with this change multicast acls will be matched by this `if` and filtered out

**Comment by npinaeva** [comment]
`OVN Egress Gateway` failure :) retest

**Pull request merged**